assets/pages/monthly-contributions/helpers/ajax.js

```diff
@@ -59,22 +61,68 @@ function requestData(paymentFieldName: PaymentField, token: string, getState: Fu
 
 }
 
-export default function postCheckout(paymentFieldName: PaymentField): Function {
-  return (token: string, dispatch: Function, getState: Function) => {
+function statusPoll(dispatch: Function, getState: Function) {
+  const state = getState();
 
-    const request = requestData(paymentFieldName, token, getState);
+  if (state.monthlyContrib.pollCount > MAX_POLLS) {
+    const url: string = addQueryParamToURL(MONTHLY_CONTRIB_PENDING, 'INTCMP', getState().intCmp);
+    window.location.assign(url);
+  }
 
-    return fetch(MONTHLY_CONTRIB_ENDPOINT, request).then((response) => {
+  dispatch(incrementPollCount());
 
-      const url: string = addQueryParamToURL(MONTHLY_CONTRIB_THANKYOU, 'INTCMP', getState().intCmp);
+  const request = {
+    method: 'GET',
+    headers: { 'Content-Type': 'application/json', 'Csrf-Token': state.csrf.token },
+    credentials: 'same-origin',
+  };
 
-      if (response.ok) {
-        window.location.assign(url);
-        return;
+  return fetch(state.monthlyContrib.trackingUri, request).then((response) => {
+    handleStatus(response, dispatch, getState); // eslint-disable-line no-use-before-define
+  });
+}
+
+function delayedStatusPoll(dispatch: Function, getState: Function) {
+  setTimeout(() => statusPoll(dispatch, getState), POLLING_INTERVAL);
+}
+
+function handleStatus(response: Response, dispatch: Function, getState: Function) {
+  const state = getState();
+  if (response.ok) {
+    response.json().then((status) => {
+      dispatch(setTrackingUri(status.trackingUri));
+      switch (status.status) {
+        case 'pending':
+          delayedStatusPoll(dispatch, getState);
+          break;
+        case 'failure':
+          dispatch(checkoutError(status.message));
+          break;
+        case 'success':
+          window.location.assign(addQueryParamToURL(MONTHLY_CONTRIB_THANKYOU, 'INTCMP', getState().intCmp));
+          break;
+        default:
+          delayedStatusPoll(dispatch, getState);
```

Comment:
"The definition of insanity is doing the same thing over and over and expecting different results."